### ISSUE TITLE
a11y issues with Vega visualizations

### ISSUE THREAD

**Issue by benhammondmusic** [SatcherInstitute/health-equity-tracker#1053]
- [x] Explore relationships between x and y in z	SR	[A11Y - Health Equity Tracker]: Graphs under Explore relationships page don't have a proper label_Web_1.1	"ENVIRONMENT:
macOS 11.3 | Chrome Version 90.0.4430.212 (Official Build) (x86_64) | Voiceover
Chrome OS Version 91.0.4472.60 (Official Build) beta (64-bit) | Chromevox
PRE REQUISITES:
Turn on the respective screen reader
STEPS TO REPRODUCE:
1). Open https://healthequitytracker.org/exploredata
2). Perform action on next button to reach the final sub page
3). Verify the options are selected as such, the page is 'Explore relationships between Diabetes and Covid-19 in United States.
4). Navigate to population profile card and expand the card
5). Linear navigate to the graphs under profile card
6). Observe the announcement.
EXPECTED RESULT:
Graphs on a page should have a label and they should be announced properly by screen reader.
OBSERVED RESULT:
Graphs (Population by race & Population by age) under population profile card don't have a label. Screen reader announces the graphs as Vega visualization visualization. Upon further navigation in mac, voiceover announces the graphs as unlabeled image.
Same issue is replicable across all graphs on the page.
Replication video attached"	https://drive.google.com/file/d/1X2WO3NmIkV9aUSzL0hTwJHAeJ-j1yB2q/view?usp=sharing	Same issue is observed on Compare rates page.

- [x] Explore relationships between x and y in z	SR	[A11Y - Health Equity Tracker]: Maps under Explore relationships page don't have a proper label_Web_1.1	"ENVIRONMENT:
macOS 11.3 | Chrome Version 90.0.4430.212 (Official Build) (x86_64) | Voiceover
Chrome OS Version 91.0.4472.60 (Official Build) beta (64-bit) | Chromevox
PRE REQUISITES:
Turn on the respective screen reader
STEPS TO REPRODUCE:
1). Open https://healthequitytracker.org/exploredata
2). Perform action on next button to reach the final sub page
3). Verify the options are selected as such, the page is 'Explore relationships between Diabetes and Covid-19 in United States.
4). Navigate through the page
5). Observe the announcement.
EXPECTED RESULT:
Maps on the page should have a label and they should be announced properly by screen reader.
OBSERVED RESULT:
Maps on the page don't have a proper label. Screen reader announces all the maps in the page as choropleth map. Upon further navigation in mac, voiceover announces the maps as unlabeled image.
Replication video attached"	https://drive.google.com/file/d/1FQ9sE5gnnIWl4aUj66_KmnX1uPOT9uB4/view?usp=sharing	Same issue is observed on Compare rates page.

- [x] Explore relationships between x and y in z	Screen Reader		[A11Y - Health Equity Tracker]: User unable to reach graphs on the page using talkback_mweb_1.2 (a)	"Device: Pixel 4
Android Version: 11
Chrome Browser Version: 91.0.4472.101
PRE REQUISITES:
Turn on Talkback (Settings > Accessibility > Talkback)
STEPS TO REPRODUCE:
1). Open https://healthequitytracker.org/exploredata
2). Linear navigate to next button and double tap till you reach the final sub page
3). Verify the options are selected as such, the page is 'Explore relationships between Diabetes and Covid-19 in United States.
4). Linear navigate profile card expand button and double tap
5). Linear navigate to graphs under profile card
6). Observe the behavior.
EXPECTED RESULT:
User should be able to reach all elements in a page using linear navigation
OBSERVED RESULT:
User is unable to reach the graphs under profile card using talkback, screen reader skips the graphs in the page. User won't have any idea that the page has graphs
Replication video - https://drive.google.com/file/d/1mYELr_rZ_vyuMjqiH2n49Brex66UzlWX/view?usp=sharing"

- [x] Explore relationships between x and y in z	Screen Reader		[A11Y - Health Equity Tracker]: User unable to reach maps on the page using talkback_mweb_1.2 (a)	"Device: Pixel 4
Android Version: 11
Chrome Browser Version: 91.0.4472.101
PRE REQUISITES:
Turn on Talkback (Settings > Accessibility > Talkback)
STEPS TO REPRODUCE:
1). Open https://healthequitytracker.org/exploredata
2). Linear navigate to next button and double tap till you reach the final sub page
3). Verify the options are selected as such, the page is 'Explore relationships between Diabetes and Covid-19 in United States.
4). Linear navigate to the maps on the page
5). Observe the behavior.
EXPECTED RESULT:
User should be able to reach all elements in a page using linear navigation
OBSERVED RESULT:
User is unable to reach the maps on the page using talkback, screen reader skips the maps. User won't have any idea that the page has maps
Replication video - https://drive.google.com/file/d/1e7o8LpYzP5L6LQXaEGbyh-4UdV8i141_/view?usp=sharing"

**Comment by benhammondmusic** [comment]
http://vis.csail.mit.edu/pubs/vis-text-model/

“This is a [chart-type] entitled [chart-title]. [y-encoding] is plotted on the vertical y-axis from [y-min] to [y-max]. [x-encoding] is plotted on the horizontal x-axis from [x-min] to [x-max].”

**Comment by benhammondmusic** [comment]
> “I am totally blind, and virtually all data visualizations I encounter are undescribed, and as such are unavailable. This has been acutely made clear on Twitter and in newspapers around the COVID-19 pandemic and the recent U.S. election. Often, visualizations are presented with very little introduction or coinciding text. I feel very left out of the world and left out of the ability to confidently traverse that world. The more data I am unable to access, the more vulnerable and devalued I feel.”

**Comment by benhammondmusic** [comment]
A majority of blind readers (63%, n=19) were emphatic that descriptions should not contain an author’s subjective interpretations, contextual information, or editorializing about the visualized data (i.e., Level 4 content). Consistent with blind readers ranking this as among the least useful (Region 4.B), BR20 succinctly articulated a common sentiment: > “I want the information to be simply laid out, not peppered with subjective commentary… I just prefer it to be straight facts, not presumptions or guesstimates.”

**Comment by benhammondmusic** [comment]
https://medium.com/nightingale/writing-alt-text-for-data-visualization-2a218ef43f81

Here’s a simple formula for writing alt text for data visualization:
> alt= “Chart type of type of data where reason for including chart” 
Include a link to data source somewhere in the text

**Comment by benhammondmusic** [comment]
https://www.wgbh.org/foundation/ncam/guidelines/line-graphs
